programs/svm-spoke/src/instructions/fill.rs

```diff
@@ -65,27 +65,9 @@ pub struct FillV3Relay<'info> {
     pub system_program: Program<'info, System>,
 }
 
-#[derive(AnchorSerialize, AnchorDeserialize, Clone)] // TODO: do we need all of these?
-
-// TODO: should we move this data structure to a common module?
-pub struct V3RelayData {
-    pub depositor: Pubkey,
-    pub recipient: Pubkey,
-    pub exclusive_relayer: Pubkey,
-    pub input_token: Pubkey,
-    pub output_token: Pubkey,
-    pub input_amount: u64,
-    pub output_amount: u64,
-    pub origin_chain_id: u64,
-    pub deposit_id: u32,
-    pub fill_deadline: u32,
-    pub exclusivity_deadline: u32,
-    pub message: Vec<u8>,
-}
-
 pub fn fill_v3_relay(
     ctx: Context<FillV3Relay>,
-    relay_hash: [u8; 32], // include in props, while not using it, to enable us to access it from the #Instruction Attribute within the accounts. This enables us to pass in the relay_hash PDA.
+    _: [u8; 32], // include in props, while not using it, to enable us to access it from the #Instruction Attribute within the accounts. This enables us to pass in the relay_hash PDA.
```

Comment:
we can delete this prop altogether in the inner instruction call as we don't necessarily need to pass it from the outer function in `svm-spoke/src/lib.rs`. Since instruction args in `FillV3Relay` accounts is passed from the outer instruction call, not the inner function. So only need to rename `relay_hash` in the outer function to `_relay_hash` and not pass it to this inner instruction function.

Reply:
ye, that's great! works as you explained. that's actually a fair bit cleaner. we still "expose" the method with this prop but the internal function does not need it as it's used by the instructions.